### PR TITLE
Add support to CIS-1.23 1.0.0

### PR DESCRIPTION
**Pull request by tengqm**
Closes: #1133

### CODE REVIEW

**Comment by mozillazg** [comment]
@tengqm Thanks for your contribution! I will get to review this PR in the next few days, thanks for your patient.

**Comment by chen-keinan** [comment]
@tengqm thanks you for this contribution !!  could you please update the following test and docs files with `cis-1.23` support:
- common_test.go
- architecture.md
- platforms.md

**Comment by tengqm** [comment]
> @tengqm thanks you for this contribution !! could you please update the following test and docs files with `cis-1.23` support:
> 
> * common_test.go
> * architecture.md
> * platforms.md

Done.

File: cmd/common_test.go [hunk 1]
```diff
@@ -237,6 +237,7 @@ func TestMapToCISVersion(t *testing.T) {
 		{kubeVersion: "1.19", succeed: true, exp: "cis-1.20"},
 		{kubeVersion: "1.20", succeed: true, exp: "cis-1.20"},
 		{kubeVersion: "1.21", succeed: true, exp: "cis-1.20"},
+		{kubeVersion: "1.21", succeed: true, exp: "cis-1.23"},
```
Comment (chen-keinan) [inline cmd/common_test.go:240]: I think unit tests are failing because of it
```suggestion
		{kubeVersion: "1.23", succeed: true, exp: "cis-1.23"},
```

**Comment by chen-keinan** [comment]
@tengqm well done!! , added another comment

File: cmd/common_test.go (hunk 1 above)
Comment (tengqm) [reply cmd/common_test.go:240]: oops ...

File: cfg/cis-1.23/master.yaml [hunk 2]
```diff
@@ -0,0 +1,959 @@
...
+              compare:
+                op: has
+                value: "DenyServiceExternalIPs"
+            - flag: "--disable-admission-plugins"
+              compare:
+                op: nothave
+                value: "DenyServiceExternalIPs"
+            - flag: "--disable-admission-plugins"
+              compare:
+                set: false
```
Comment (mozillazg) [inline cfg/cis-1.23/master.yaml:348]: The `DenyServiceExternalIPs` admission plugin is disabled by default.

```suggestion
```

**Review by mozillazg — CHANGES_REQUESTED** [review]
Thanks for your work! Added some comments.

File: docs/platforms.md [hunk 3]
```diff
@@ -6,14 +6,15 @@ Most of our supported benchmarks are defined in the [CIS Kubernetes Benchmarks](
...
-| CIS | [EKS 1.0.1](https://workbench.cisecurity.org/benchmarks/6041) | eks-1.0.1 | EKS |
-| CIS | [ACK 1.0.0](https://workbench.cisecurity.org/benchmarks/6467) | ack-1.0 | ACK |
-| CIS | [AKS 1.0.0](https://workbench.cisecurity.org/benchmarks/6347) | aks-1.0 | AKS |
-| RHEL | RedHat OpenShift hardening guide | rh-0.7 | OCP 3.10-3.11 |
-| CIS | [OCP4 1.1.0](https://workbench.cisecurity.org/benchmarks/6778) | rh-1.0 | OCP 4.1- |
+|------|----------------------------------------------------------------|-----------|-------------|
+| CIS  | [1.5.1](https://workbench.cisecurity.org/benchmarks/4892)      | cis-1.5   | 1.15        |
+| CIS  | [1.6.0](https://workbench.cisecurity.org/benchmarks/4834)      | cis-1.6   | 1.16-1.18   |
+| CIS  | [1.20](https://workbench.cisecurity.org/benchmarks/6246)       | cis-1.20  | 1.19-1.20   |
+| CIS  | [1.23](https://workbench.cisecurity.org/benchmarks/7532)       | cis-1.23  | 1.23        |
```
Comment (mozillazg) [inline docs/platforms.md:13]:
```suggestion
| CIS  | [1.23](https://workbench.cisecurity.org/benchmarks/7532)       | cis-1.23  | 1.22-1.23   |
```

File: cfg/cis-1.23/master.yaml (hunk 2 above)
Comment (tengqm) [reply cfg/cis-1.23/master.yaml]: After rereading the logic, I come up with a different way to express this constraint, please review.

Comment (tengqm) [reply cfg/cis-1.23/master.yaml]: done

**Review by mozillazg — CHANGES_REQUESTED** [review]

File: cfg/cis-1.23/node.yaml [hunk 4]
```diff
@@ -0,0 +1,462 @@
+---
+controls:
+version: "cis-1.20"
```
Comment (mozillazg) [inline cfg/cis-1.23/node.yaml:3]:
```suggestion
version: "cis-1.23"
```

File: cfg/cis-1.23/node.yaml [hunk 5]
```diff
@@ -0,0 +1,462 @@
...
+          chown root:root $kubeletconf
+        scored: true
+
+  - id: 4.2
+    text: "Kubelet"
+    checks:
+      - id: 4.2.1
+        text: "Ensure that the --anonymous-auth argument is set to false (Automated)"
+        audit: "/bin/ps -fC $kubeletbin"
+        audit_config: "/bin/cat $kubeletconf"    # TODO(Qiming): This is interesting
```
Comment (mozillazg) [inline cfg/cis-1.23/node.yaml:162]: It'll check the config file if not found in the command-line arguments

```suggestion
        audit_config: "/bin/cat $kubeletconf"
```

File: cfg/cis-1.23/node.yaml [hunk 6]
```diff
@@ -0,0 +1,462 @@
...
+          set the below parameters in KUBELET_CERTIFICATE_ARGS variable.
+          --tls-cert-file=<path/to/tls-certificate-file>
+          --tls-private-key-file=<path/to/tls-key-file>
+          Based on your system, restart the kubelet service. For example,
+          systemctl daemon-reload
+          systemctl restart kubelet.service
+        scored: false
+
+      - id: 4.2.11
+        text: "Ensure that the --rotate-certificates argument is not set to false (Manual)"
```
Comment (mozillazg) [inline cfg/cis-1.23/node.yaml:390]: image

```suggestion
        text: "Ensure that the --rotate-certificates argument is not set to false (Automated)"
```

File: cfg/cis-1.23/node.yaml [hunk 7]
```diff
@@ -0,0 +1,462 @@
...
+          If using a Kubelet config file, edit the file to add the line `rotateCertificates` to `true` or
+          remove it altogether to use the default value.
+          If using command line arguments, edit the kubelet service file
+          $kubeletsvc on each worker node and
+          remove --rotate-certificates=false argument from the KUBELET_CERTIFICATE_ARGS
+          variable.
+          Based on your system, restart the kubelet service. For example,
+          systemctl daemon-reload
+          systemctl restart kubelet.service
+        scored: false
```
Comment (mozillazg) [inline cfg/cis-1.23/node.yaml:414]:
```suggestion
        scored: true
```

**Review by mozillazg — CHANGES_REQUESTED** [review]

**Comment by mozillazg** [comment]
@tengqm Added some comments. Please check again. Thanks!

**Review by chen-keinan — APPROVED** [review]

File: cfg/cis-1.23/master.yaml [hunk 8]
```diff
@@ -0,0 +1,955 @@
...
+        tests:
+          bin_op: and
+          test_items:
+            - flag: "--enable-admission-plugins"
+              compare:
+                op: nothave
+                value: "DenyServiceExternalIPs"
+            - flag: "--enable-admission-plugins"
+              compare:
+                set: false
```
[the remark this answers is not in the document]
Comment (mozillazg) [reply cfg/cis-1.23/master.yaml]: This issue still exists.

File: cfg/cis-1.23/node.yaml (hunk 7 above)
Comment (mozillazg) [reply cfg/cis-1.23/node.yaml:414]: This issue still exists too.

File: cfg/cis-1.23/master.yaml [hunk 9]
```diff
@@ -0,0 +1,955 @@
...
+        tests:
+          bin_op: or
+          test_items:
+            - flag: "--enable-admission-plugins"
+              compare:
+                op: nothave
+                value: "DenyServiceExternalIPs"
+            - flag: "--enable-admission-plugins"
+              compare:
+                set: false
```
Comment (mozillazg) [inline cfg/cis-1.23/master.yaml:344]: Missed this sugesstion:

```suggestion
            - flag: "--enable-admission-plugins"
              set: false
```

**Review by mozillazg — CHANGES_REQUESTED** [review]

**Comment by mozillazg** [comment]
@tengqm There are two suggestions that were missed in your last changes. Please check again.

**Comment by tengqm** [comment]
> @tengqm There are two suggestions that were missed in your last changes. Please check again. Thanks for your patience.

Please check again. Thanks.

**Review by mozillazg — APPROVED** [review]
LGTM, Thanks for your contribution!

**Comment by chen-keinan** [comment]
@tengqm Last thing , please rebase your branch with upstream so I can merge it.

**Comment by tengqm** [comment]
> @tengqm Last thing , please rebase your branch with upstream so I can merge it.

Rebased onto main.

**Pull request merged**